File: src/main/java/com/tomatokey/framework/configuration/mvc/auth/NonAuthorize.java
package com.tomatokey.framework.configuration.mvc.auth;

import org.springframework.web.bind.annotation.RestController;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * {@link RestController}アノテーションが付与されたクラス内の
 * メソッドに付与すると認証の対象外にすることができます
 */
@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface NonAuthorize {
}

File: src/main/kotlin/com/prototype/framework/configuration/mvc/auth/NonAuthorize.kt
package com.prototype.framework.configuration.mvc.auth

/**
 * [RestController]アノテーションが付与されたクラス内の
 * メソッドに付与すると認証の対象外にすることができます
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class NonAuthorize 